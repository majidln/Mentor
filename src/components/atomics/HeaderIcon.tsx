import React, {FC, ReactElement} from 'react';
import {
  TouchableOpacity,
  Text,
  StyleSheet,
  TouchableOpacityProps,
} from 'react-native';

export interface HeaderIconProps extends TouchableOpacityProps {
  title?: string;
}

export const HeaderIcon: FC<HeaderIconProps> = ({
  title = '',
  ...rest
}): ReactElement => {
  return (
    <TouchableOpacity style={styles.wrapper} {...rest}>
      <Text>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    height: '100%',
    paddingHorizontal: 8,
    justifyContent: 'center',
  },
});
