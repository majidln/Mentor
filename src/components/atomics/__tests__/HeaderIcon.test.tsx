import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

import {HeaderIcon} from './../HeaderIcon';

describe('Test for atomic HeaderIcon', () => {
  const title = 'Add';

  it('HeaderIcon render correctly', async () => {
    const {findByTestId} = render(
      <HeaderIcon
        title={title}
        onPress={() => null}
        testID="add_header_icon"
      />,
    );

    const button = await findByTestId('add_header_icon');
    fireEvent.press(button);
  });

  it('HeaderIcon renders correctly', () => {
    const tree = renderer
      .create(<HeaderIcon title={title} onPress={() => null} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
