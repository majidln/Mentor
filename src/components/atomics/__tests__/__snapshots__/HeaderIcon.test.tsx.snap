// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for atomic HeaderIcon HeaderIcon renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": "100%",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
    }
  }
>
  <Text>
    Add
  </Text>
</View>
`;
